src/BlazorWebView/src/SharedSource/BlazorWebViewServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+#if WEBVIEW2_WINFORMS
+using Microsoft.AspNetCore.Components.WebView.WindowsForms;
+#elif WEBVIEW2_WPF
+using Microsoft.AspNetCore.Components.WebView.Wpf;
+#elif WEBVIEW2_MAUI
+using Microsoft.AspNetCore.Components.WebView.Maui;
+using Microsoft.Maui.Hosting;
+#else
+#error Must define WEBVIEW2_WINFORMS, WEBVIEW2_WPF, WEBVIEW2_MAUI
+#endif
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+	/// <summary>
+	/// Extension methods to <see cref="IServiceCollection"/>.
+	/// </summary>
+	public static class BlazorWebViewServiceCollectionExtensions
+	{
+		/// <summary>
+		/// Configures <see cref="IServiceCollection"/> to add support for <see cref="BlazorWebView"/>.
+		/// </summary>
+		/// <param name="services">The <see cref="IServiceCollection"/>.</param>
+		/// <returns>The <see cref="IServiceCollection"/>.</returns>
+#if WEBVIEW2_WINFORMS
+		public static IServiceCollection AddWindowsFormsBlazorWebView(this IServiceCollection services)
+#elif WEBVIEW2_WPF
+		public static IServiceCollection AddWpfBlazorWebView(this IServiceCollection services)
+#elif WEBVIEW2_MAUI
+		public static IServiceCollection AddMauiBlazorWebView(this IServiceCollection services)
+#else
+#error Must define WEBVIEW2_WINFORMS, WEBVIEW2_WPF, WEBVIEW2_MAUI
+#endif
+		{
+			services.AddBlazorWebView();
+			services.TryAddSingleton(new BlazorWebViewDeveloperTools { Enabled = false });
+#if WEBVIEW2_WINFORMS
+#elif WEBVIEW2_WPF
+#elif WEBVIEW2_MAUI
+			services.ConfigureMauiHandlers(static handlers => handlers.AddHandler<IBlazorWebView, BlazorWebViewHandler>());
+#else
+#error Must define WEBVIEW2_WINFORMS, WEBVIEW2_WPF, WEBVIEW2_MAUI
+#endif
```

Comment:
I believe the general pattern we've used for platform specific is to just have the ifdef for the specific platform at hand:

```suggestion
#if WEBVIEW2_MAUI
			services.ConfigureMauiHandlers(static handlers => handlers.AddHandler<IBlazorWebView, BlazorWebViewHandler>());
#endif
```

Reply:
It's either or, tbh. It's more correct to check for all the cases and error out on an else, as it errors out explicitly instead of silently succeeding.